homeassistant/components/binary_sensor/mqtt.py

```diff
@@ -39,6 +41,8 @@
     vol.Optional(CONF_PAYLOAD_ON, default=DEFAULT_PAYLOAD_ON): cv.string,
     vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
     vol.Optional(CONF_FORCE_UPDATE, default=DEFAULT_FORCE_UPDATE): cv.boolean,
+    vol.Optional(CONF_OFF_DELAY):
+    vol.Any(cv.time_period, cv.positive_timedelta),
```

Comment:
Shouldn't this be `vol.All`? I mean first it needs to be converted to a timedelta, *and then* the timedelta needs to be checked if it's positive.

Reply:
Fixed.